### DISCUSSION TITLE
Manipulation of symbolic expressions

### DISCUSSION THREAD

**Discussion by QMechanic** [Z3Prover/z3#6502] [General]
Is there any functionality to manipulate expressions. Similar to computer algebra systems?

I'm not asking for an advanced simplify() function. Although that would be great.
I'm just looking for an expand() function.

From what I can find. There is no possible way to expand equations like
`(a+b)*(c+d+e)` -> `a*c + a*d + a*e + b*c + b*d + b*e `

I've been unable to find any methods to expand equations like this.
Are there any tactics which can do this manipulation of expressions?

Thanks in advance.

**Comment by NikolajBjorner** [comment]
https://microsoft.github.io/z3guide/docs/strategies/summary/#tactic-simplify
